src/software/new_geom/circle.cpp

```diff
@@ -0,0 +1,53 @@
+#include "software/new_geom/circle.h"
+
+Circle::Circle() : origin_(Point()), radius_(0) {}
+
+Circle::Circle(const Point &origin, double radius) : origin_(origin)
+{
+    setRadius(radius);
+}
+
+void Circle::setOrigin(const Point &o)
+{
+    origin_ = o;
+}
+
+Point Circle::getOrigin() const
+{
+    return Point(origin_);
+}
+
+void Circle::setRadius(double radius)
+{
+    if (radius < 0)
+    {
+        throw std::invalid_argument("Circle radius cannot be negative");
```

Comment:
Could be useful to show what the invalid argument was

Reply:
Fixed